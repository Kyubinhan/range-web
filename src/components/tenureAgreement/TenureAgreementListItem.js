import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'semantic-ui-react';

const propTypes = {
  tenureAgreement: PropTypes.object.isRequired,
  onViewClicked: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
  isActive: PropTypes.bool.isRequired,
};

const defaultProps = {
  tenureAgreement: {},
};

class TenureAgreementListItem extends Component {
  state = {
    isOpen: false
  }

  onViewClicked = (e) => {
    e.preventDefault();

    const { index, onViewClicked } = this.props;
    onViewClicked(index)
  }

  render() {
    const { tenureAgreement, isActive } = this.props;

    return (
      <li className="tenure-agreement-list-item">
        <div className="tenure-agreement-list-item__content">
          <div className="tenure-agreement-list-item__content__left">
            <div className="tenure-agreement-list-item__content__number">
              {`RAN ${tenureAgreement.number}`}
            </div>

            <div className="tenure-agreement-list-item__content__divider" />
            <div className="tenure-agreement-list-item__content__info">
              {tenureAgreement.tenureHolder.name}
            </div>
          </div>

          <div className="tenure-agreement-list-item__content__right">
            <Button 
              primary
              onClick={this.onViewClicked}
            >
              View
            </Button>
          </div>
        </div>

        <div 
          className={"tenure-agreement-list-item__collapse" + (isActive ? " tenure-agreement-list-item__collapse--active" : "")}
        >
          hello bottom!
        </div>
      </li>
    );
  }
}

TenureAgreementListItem.propTypes = propTypes;
TenureAgreementListItem.defaultProps = defaultProps;
export default TenureAgreementListItem;