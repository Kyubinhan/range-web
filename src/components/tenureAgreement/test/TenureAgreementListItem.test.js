import React from 'react';
import { shallow } from 'enzyme';
import TenureAgreementListItem from '../TenureAgreementListItem';
import { getMockTenureAgreement } from './mockValues';

const props = {};
const setupProps = () => {
  props.tenureAgreement = getMockTenureAgreement(1);
  props.onViewClicked = jest.fn();
  props.index = 1;
  props.isActive = false;
};

beforeEach(() => {
  setupProps();
});

describe('TenureAgreementListItem', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TenureAgreementListItem {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});