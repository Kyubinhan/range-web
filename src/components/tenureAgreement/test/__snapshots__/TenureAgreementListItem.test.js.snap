// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TenureAgreementListItem renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(TenureAgreementListItem)
    index={1}
    isActive={false}
    onViewClicked={[Function]}
    tenureAgreement={
        Object {
            "id": 1,
            "number": 1,
            "region": "Victoria Ranch 1",
            "status": "pending",
            "tenureHolder": Object {
              "name": "tenure holder 1",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "render": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "render": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
